Test failure for RegexStringGeneratorTests.shouldCorrectlyReplaceCharacterGroups

## Bug Report

### Steps to Reproduce:
 1. Run the RegexStringGeneratorTests.shouldCorrectlyReplaceCharacterGroups test via intelliJ

### Expected Result:
Test to pass

### Actual Result:
```
java.lang.AssertionError: 
Expected: "0"
     but: was null
```

This test (amongst others) aren't running with our pull requests see #556

--- a/generator/src/test/java/com/scottlogic/deg/generator/generation/RegexStringGeneratorTests.java
+++ b/generator/src/test/java/com/scottlogic/deg/generator/generation/RegexStringGeneratorTests.java
@@ -7,10 +7,9 @@ import org.junit.Assert;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.List;
+import java.util.*;
 import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
 
 import static org.hamcrest.CoreMatchers.not;
 import static org.hamcrest.Matchers.*;
@@ -318,7 +317,14 @@ public class RegexStringGeneratorTests {
     private void expectFirstResult(String expectedValue) {
         StringGenerator generator = constructGenerator(true);
 
-        String actualValue = generator.generateAllValues().iterator().next();
+        String actualValue = StreamSupport
+            .stream(
+                Spliterators.spliteratorUnknownSize(
+                    generator.generateAllValues().iterator(),
+                    Spliterator.ORDERED),
+                false)
+            .limit(1)
+            .findFirst().orElse(null);
 
         Assert.assertThat(
                 actualValue,

